gst/nnstreamer/elements/gsttensor_transform.c

```diff
@@ -2041,13 +2041,10 @@ gst_tensor_transform_transform_caps (GstBaseTransform * trans,
       gst_caps_append (result_aux, gst_tensor_caps_from_config (&out_config));
     } else {
       gst_caps_append (result_aux, gst_tensors_caps_from_config (&out_config));
-    }
 
-    /* remove `types` field from caps */
-    if (is_types_not_fixed) {
-      GstStructure *s;
-      for (j = 0; j < gst_caps_get_size (result_aux); ++j) {
-        s = gst_caps_get_structure (result_aux, j);
+      /* remove `types` field from caps */
+      if (is_types_not_fixed) {
+        GstStructure *s = gst_caps_get_structure (result_aux, 0);
```

Comment:
How do you sure that result_aux will have only one GstStructure?

Reply:
The caps 'result_aux' is created in the loop (line 2010), and new capability is appended once (2040~2043).